examples/google-tables/convert.ts

```diff
@@ -0,0 +1,430 @@
+    case "boolean":
+      return new BooleanColumnConverter(context, column, table);
+    case "date":
+      if (!column.dateDetails?.hasTime) {
+        return new DateColumnConverter(context, column, table);
+      }
+      return new DateTimeColumnConverter(context, column, table);
+    case "person":
+    case "creator":
+    case "updater":
+      return new PersonColumnConverter(context, column, table);
+    case "person_list":
+      let personConverter = new PersonColumnConverter(context, column, table);
+      if (column.multipleValuesDisallowed) {
+        // We want to display this as a single person selector in Coda, but the
+        // value come back from the API as a list of one.
+        return new UnwrapColumnConverter(personConverter);
+      }
+      return new ListColumnConverter(personConverter);
+    case "dropdown":
+      return new SelectListColumnConverter(context, column, table);
+    case "check_list":
+    case "tags_list":
+      return new MultiSelectListColumnConverter(context, column, table);
+    case "drive_attachment_list":
+      return new DriveFilesColumnConverter(context, column, table);
+    case "file_attachment_list":
+      return new FilesColumnConverter(context, column, table);
+    case "location":
+      return new LocationColumnConverter(context, column, table);
+    case "relationship":
+      return new RelationshipColumnConverter(context, column, table);
+    case "create_timestamp":
+    case "update_timestamp":
+    case "comment_timestamp":
+      return new TimestampColumnConverter(context, column, table);
+    default:
+      if (column.dataType.endsWith("_list")) {
+        // Handle lists of basic types.
+        let baseType = column.dataType.slice(0, -5);
+        let baseColumn: Column = {...column, dataType: baseType};
+        let baseConverter = getConverter(context, baseColumn, table);
+        if (baseConverter) {
+          return new ListColumnConverter(baseConverter);
+        }
+      }
+      // eslint-disable-next-line no-console
+      console.error(`No converter found for column type: ${column.dataType}`);
+      return new UnknownColumnConverter(context, column, table);
+  }
+}
+
+// Abstract class that all converter classes extend.
+abstract class ColumnConverter<T, C> {
+  column: Column;
+  table: Table;
+  context: coda.ExecutionContext;
+
+  constructor(context: coda.ExecutionContext, column: Column, table: Table) {
+    this.context = context;
+    this.column = column;
+    this.table = table;
+  }
+
+  // Each implementation must define the property schema.
+  abstract getSchema(): coda.Schema & coda.ObjectSchemaProperty;
+
+  // Default to passing through the value as-is, in both directions.
+  formatValueForSchema(value: T): C {
+    return value as any;
+  }
+  formatValueForApi(value: C): T {
+    return value as any;
+  }
+}
+
+class TextColumnConverter extends ColumnConverter<string, string> {
+  getSchema() {
+    return coda.makeSchema({
```

Comment:
Not sure how renaming in Google Tables works, but you might consider including a `fixedId` on these schemas so if a rename does take place on the Tables side, you don't break the Coda table.

Reply:
Yep, this is already done in `helpers.ts` in `getPropertySchema()`. Not done in the individual converters since I don't want to repeat that code.